docs/proposals/petset.md

```diff
@@ -0,0 +1,533 @@
+            }
+        }]
+    }
+}
+
+// Proposal 2b: append identity to specified field paths
+// Downside: Flexibility is limited, may need "operations" on the field in order to transform value
+{
+    "spec": {
+        // ...
+
+        "supportTemplates": [{
+            "persistentVolumeClaim": {
+                "metadata": {"name": "pod-name-pvc-"}
+            },
+            "fieldPaths": ["metadata.name"] // controller appends IDENTTY to field metadata.name
+        }]
+    }
+}
+
+// Proposal 2c: append identity to specified field paths, name is special
+// Downside: Flexibility is limited, transformations may be required
+{
+    "spec": {
+        // ...
+
+        "supportTemplates": [{
+            "configData": {
+                "metadata": {"generateName": "pod-name-pvc-"}
+                // controller appends IDENTITY to field metadata.generateName and sets metadata.name from that
+            },
+            "fieldPaths": ["data[identity]"] // controller appends IDENTITY to the "identity" key of config data
+        }]
+    }
+}
+
+// Proposal 2d: support template transformation of a field path
+// Downside: Unrestricted use possibly too powerful
+// Downside: Harder for clients to recognize what fields are parameterized
+{
+    "spec": {
+        "template": {
+            "spec": {
+                "volumes": [{
+                    "persistentVolumeClaim": {
+                        "claimName": "pod-name-pvc" // controller appends IDENTITY to this claim automatically
+                    }
+                }]
+            }
+        },
+        "supportTemplates": [{
+            "persistentVolumeClaim": {
+                "metadata": {"name": "pod-name-pvc-$(IDENTITY)"} // controller adds IDENTTY to this claim as defined
+            }
+        }]
+    }
+}
+
+// Proposal 2e: specify merge-patch semantics with some variable replacement
+{
+    "spec": {
+        // ...
+
+        "supportTemplates": [{
+            "persistentVolumeClaim": {
+                "spec": {...}
+            },
+            "transform": [{"op": "add", "path": "/metadata/name", "value": "pod-name-pvc-$(IDENTITY)"}]
+        }]
+    }
+}
+
+```
+
+Of all proposals, 2e is the closest to actual semantics we support today (variable replacement in EnvVar and
+patch operations). The path syntax from RFC6902 is not optimal (we prefer JQ field path semantics), but we
+could consider supporting both. Since we desire to support an integer value for quorum size, we may need to
+transform the set size value to an integer for replacement into a supporting resource that needs that value
+as a number:
+
+```
+// Proposal 2f: variant: specify merge-patch semantics with some variable replacement as an integer
+{
+    "spec": {
+        // ...
+
+        "supportTemplates": [{
+            "replicationController": {
+                "spec": {"replicas": 1}
+            },
+            // "$(REPLICAS)" is transformed into 5, not "5"
```

Comment:
I don't understand this proposal. REPLICAS is from the PetSet or from the RC support template? If the latter, would all spec fields be automatically recognized?

Is this example derived from an actual use case?